requirements.txt

```diff
@@ -1,7 +1,9 @@
+aiohttp>=3.8.1
```

Comment:
If memory serves, it is necessary.  Importing `fsspec` alone (which is necessary) doesn't bring this is in, and this is how remote data fetching works.  As an example:

```bash
python3 -m venv venv
source venv/bin/activate
pip install zarr
python3
```
and then:
```python
>>> import zarr
>>> zarr.open('https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.1/6001240.zarr')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/zarr/convenience.py", line 82, in open
    _store: BaseStore = normalize_store_arg(
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/zarr/storage.py", line 114, in normalize_store_arg
    return FSStore(store, mode=mode, **(storage_options or {}))
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/zarr/storage.py", line 1117, in __init__
    import fsspec
ModuleNotFoundError: No module named 'fsspec
```
And then after quitting out of the above python terminal
```bash
pip install fsspec
```
and then
```python
>>> import zarr
>>> zarr.open('https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.1/6001240.zarr')
Traceback (most recent call last):
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/fsspec/registry.py", line 219, in get_filesystem_class
    register_implementation(protocol, _import_class(bit["class"]))
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/fsspec/registry.py", line 242, in _import_class
    mod = importlib.import_module(mod)
  File "/usr/local/Caskroom/miniconda/base/envs/batchglm/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/fsspec/implementations/http.py", line 11, in <module>
    import aiohttp
ModuleNotFoundError: No module named 'aiohttp'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/zarr/convenience.py", line 82, in open
    _store: BaseStore = normalize_store_arg(
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/zarr/storage.py", line 114, in normalize_store_arg
    return FSStore(store, mode=mode, **(storage_options or {}))
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/zarr/storage.py", line 1125, in __init__
    self.map = fsspec.get_mapper(url, **storage_options)
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/fsspec/mapping.py", line 228, in get_mapper
    fs, urlpath = url_to_fs(url, **kwargs)
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/fsspec/core.py", line 403, in url_to_fs
    cls = get_filesystem_class(protocol)
  File "/Users/ilangold/Projects/Theis/batchglm/venv/lib/python3.8/site-packages/fsspec/registry.py", line 221, in get_filesystem_class
    raise ImportError(bit["err"]) from e
ImportError: HTTPFileSystem requires "requests" and "aiohttp" to be installed
```
And then you need to install `aiohttp` - I suspect that `requests` is already a dependent of one of the packages listed here since I would have added that if needed, but it's possible I forgot.

Reply:
Ok, thanks for diving into this. As for requests, we can safely omit them, I know `omnipath` requires them.